sql/functions.sql

```diff
@@ -515,36 +628,38 @@ CREATE OR REPLACE FUNCTION add_location(
     keywords INTEGER[],
     rank_search INTEGER,
     rank_address INTEGER,
+    in_postcode TEXT,
     geometry GEOMETRY
   ) 
   RETURNS BOOLEAN
   AS $$
 DECLARE
   locationid INTEGER;
-  isarea BOOLEAN;
   centroid GEOMETRY;
   diameter FLOAT;
   x BOOLEAN;
   splitGeom RECORD;
   secgeo GEOMETRY;
+  postcode TEXT;
```

Comment:
I couldn't say what the length limit is for postcodes world-wide. Given that there is no difference in performance or storage space between TEXT and VARCHAR() in Postgresql, there is not reason to set an artificial limit.